allure-cucumber5-jvm/src/main/java/io/qameta/allure/cucumber5jvm/testsourcemodel/TestSourcesModel.java

```diff
@@ -0,0 +1,209 @@
+package io.qameta.allure.cucumber5jvm.testsourcemodel;
+
+import gherkin.AstBuilder;
+import gherkin.Parser;
+import gherkin.ParserException;
+import gherkin.TokenMatcher;
+import gherkin.ast.Background;
+import gherkin.ast.Examples;
+import gherkin.ast.Feature;
+import gherkin.ast.GherkinDocument;
+import gherkin.ast.Node;
+import gherkin.ast.ScenarioDefinition;
+import gherkin.ast.ScenarioOutline;
+import gherkin.ast.Step;
+import gherkin.ast.TableRow;
+import io.cucumber.core.exception.CucumberException;
```

Comment:
Ideally you don't depend on anything from `cucumber-core`. You should only need `io.cucumber.plugin`.

And please note that classes that are safe to use are annotated with `@org.apiguardian.api.API(status = API.Status.STABLE)`.

Reply:
I would suggest replacing the `CucumberException` with ra egular `IllegalStateException`.

Reply:
Noted. Will remove CucumberException and any dependency on cucumber-core.